dtmsvr/trans_class.go

```diff
@@ -19,11 +19,20 @@ import (
 	"github.com/gin-gonic/gin"
 )
 
+// triggerType trigger transaction type
+type triggerType int
+
+const (
+	triggerManual triggerType = iota //triggerManual manual trigger
+	triggerCron                      //triggerCron cron trigger
+)
+
 // TransGlobal global transaction
 type TransGlobal struct {
 	storage.TransGlobalStore
 	lastTouched      time.Time // record the start time of process
 	updateBranchSync bool
+	triggerType      triggerType
```

Comment:
triggerType的设计是因为什么？是必要的的吗？工作原理是什么样？

Reply:
因为在 transMsgProcessor.ProcessOnce 中需要排除掉 Cron 触发的。 
```go
	if cmc.Delay > 0 && t.triggerType == triggerManual {
		t.delayCronTime(cmc.Delay)
		return nil
	}
```

Reply:
参考一下isTimeout，采用时间判断，貌似代码更少，还会更直观一些

Reply:
我了解一下

Reply:
加入了 needDelay 来判断，确实这样更加直观，并且兼顾的场景更加多
```go
func (t *TransGlobal) needDelay(delay uint64) bool {
	return time.Since(*t.CreateTime)+CronForwardDuration < time.Duration(delay)*time.Second
}
```